<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CenterlineMetrics</class>
 <widget class="qMRMLWidget" name="CenterlineMetrics">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>460</width>
    <height>1006</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="parametersCollapsibleButton">
     <property name="locale">
      <locale language="English" country="UnitedStates"/>
     </property>
     <property name="text">
      <string>Parameters</string>
     </property>
     <layout class="QFormLayout" name="parametersFormLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="inputCenterLineLabel">
        <property name="text">
         <string>Input centerline:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="inputModelSelector">
        <property name="toolTip">
         <string>Pick the input VMTK centerline.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLModelNode</string>
          <string>vtkMRMLMarkupsCurveNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="editEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="outputTableLabel">
        <property name="text">
         <string>Output table:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="outputTableSelector">
        <property name="toolTip">
         <string>Pick the output table to the algorithm.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLTableNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="outputPlotSeriesLabel">
        <property name="text">
         <string>Output plot series:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLNodeComboBox" name="outputPlotSeriesSelector">
        <property name="toolTip">
         <string>Pick the output plot series to the algorithm.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLPlotSeriesNode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="actionGroupBox">
     <property name="locale">
      <locale language="English" country="UnitedStates"/>
     </property>
     <property name="title">
      <string/>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="toggleLayoutButton">
        <property name="toolTip">
         <string>Toggle between the plot layout and the preceding one.</string>
        </property>
        <property name="text">
         <string>Toggle layout</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="applyButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Run the algorithm.</string>
        </property>
        <property name="text">
         <string>Apply</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="advancedCollapsibleButton">
     <property name="text">
      <string>Advanced</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="horizontalSpacing">
       <number>4</number>
      </property>
      <property name="verticalSpacing">
       <number>4</number>
      </property>
      <property name="leftMargin">
       <number>4</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>4</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item row="0" column="1">
       <widget class="QFrame" name="moveGroupBox">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="moveSliceViewLabel">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>Move slice view:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="qMRMLNodeComboBox" name="sliceViewSelector">
           <property name="toolTip">
            <string>Move this slice view to a point along the centerline</string>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLSliceNode</string>
            </stringlist>
           </property>
           <property name="addEnabled">
            <bool>false</bool>
           </property>
           <property name="removeEnabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="moveToPointLabel">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>To point:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="ctkSliderWidget" name="moveToPointSliderWidget">
           <property name="toolTip">
            <string>Point of the centerline where the selected slice view should be moved</string>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="decimals">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="jumpCentredInSliceNodeCheckBox">
           <property name="toolTip">
            <string>Centre the selected point in the slice view</string>
           </property>
           <property name="text">
            <string>Centre in slice view</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="distanceLabel">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>Distance:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="distanceValueLabel">
           <property name="toolTip">
            <string>Cumulative distance from start of centerline</string>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="diameterLabel">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>Diameter (MIS):</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLabel" name="diameterValueLabel">
           <property name="toolTip">
            <string>Maximum inscribed sphere (MIS) diameter at selected point.</string>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="coordinatesLabel">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>Coordinates:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLabel" name="coordinatesValueLabel">
           <property name="toolTip">
            <string>RAS coordinate of selected point</string>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="scaledContents">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QGroupBox" name="moveHelperGroupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="title">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QPushButton" name="moveToMinimumPushButton">
           <property name="toolTip">
            <string>Move the selected slice view to the point of smallest diameter</string>
           </property>
           <property name="text">
            <string>Move to minimum</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="moveToMaximumPushButton">
           <property name="toolTip">
            <string>Move the selected slice view to the point of largest diameter</string>
           </property>
           <property name="text">
            <string>Move to maximum</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="ctkCollapsibleButton" name="segmentationCollapsibleButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Select a segment to view the cross-section and its surface area. It should be the segment from which the centerline was computed.

Caution : surface area at bifurcations may not have clinical meaning.</string>
        </property>
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="text">
         <string>Cross-section area measurement</string>
        </property>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="segmentationLabel">
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>Surface:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="qMRMLNodeComboBox" name="segmentationSelector">
           <property name="toolTip">
            <string>Select an input segmentation or model node.

It should be a node from which a centerline has been computed.</string>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLSegmentationNode</string>
             <string>vtkMRMLModelNode</string>
            </stringlist>
           </property>
           <property name="noneEnabled">
            <bool>true</bool>
           </property>
           <property name="addEnabled">
            <bool>false</bool>
           </property>
           <property name="removeEnabled">
            <bool>false</bool>
           </property>
           <property name="editEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="qMRMLSegmentSelectorWidget" name="segmentSelector">
           <property name="toolTip">
            <string>Select an input segment node</string>
           </property>
           <property name="segmentationNodeSelectorVisible">
            <bool>false</bool>
           </property>
           <property name="selectNodeUponCreation">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="surfaceAreaLabel">
           <property name="text">
            <string>Cross-sectional area:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="surfaceAreaValueLabel">
           <property name="toolTip">
            <string>Area of the cross-section</string>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="derivedDiameterLabel">
           <property name="text">
            <string>Diameter (CE):</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLabel" name="derivedDiameterValueLabel">
           <property name="toolTip">
            <string>Circular equivalent (CE) diameter : that of a circle having the  surface area of the cross-section.

The absolute and percent difference with the maximum inscribed sphere (MIS) diameter are also provided.

Caution : values at bifurcations may not have clinical meaning.</string>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QPushButton" name="showAvailableCrossSectionsButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Show all available cross-sections. It may not be a complete stack, but only what has been created so far.</string>
           </property>
           <property name="text">
            <string>Show available cross-sections</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="ctkCollapsibleButton" name="coordinatesCollapsibleButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Select type of output coordinates</string>
        </property>
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="text">
         <string>Coordinates</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QGroupBox" name="coordinatesGroupBox">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="title">
            <string/>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <property name="spacing">
             <number>4</number>
            </property>
            <property name="leftMargin">
             <number>4</number>
            </property>
            <property name="topMargin">
             <number>4</number>
            </property>
            <property name="rightMargin">
             <number>4</number>
            </property>
            <property name="bottomMargin">
             <number>4</number>
            </property>
            <item>
             <widget class="QRadioButton" name="radioRAS">
              <property name="toolTip">
               <string>Right-Anterior-Superior coordinate system. Used in the Slicer scene.</string>
              </property>
              <property name="text">
               <string>RAS</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioLPS">
              <property name="toolTip">
               <string>Left-Posterior-Superior coordinate system. Used commonly in files.</string>
              </property>
              <property name="text">
               <string>LPS</string>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="distinctColumnsCheckBox">
              <property name="toolTip">
               <string>Either one column with an array of values, or a distinct column for each value of the triplet.</string>
              </property>
              <property name="text">
               <string>Separate column for each axis</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSegmentSelectorWidget</class>
   <extends>qMRMLWidget</extends>
   <header>qMRMLSegmentSelectorWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>CenterlineMetrics</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputModelSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CenterlineMetrics</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>outputTableSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CenterlineMetrics</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>outputPlotSeriesSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>282</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CenterlineMetrics</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>sliceViewSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>308</y>
    </hint>
    <hint type="destinationlabel">
     <x>270</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CenterlineMetrics</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>segmentationSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>229</x>
     <y>394</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>704</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CenterlineMetrics</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>segmentSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>229</x>
     <y>394</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>746</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>segmentationSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>segmentSelector</receiver>
   <slot>setCurrentNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>291</x>
     <y>707</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>749</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
